### PR TITLE
Refactored Command and Query class using Action class

### PR DESCRIPTION
**Pull request by hpohekar**

### CODE REVIEW

File: src/ansys/fluent/core/solver/flobject.py [hunk 1]
```diff
@@ -679,56 +679,62 @@ def _get_new_keywords(obj, kwds):
     return newkwds
 
 
-class Command(Base):
-    """Command object."""
+class IBase(Base):
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/solver/flobject.py:682]: The I prefix generally means interface so your intention is lost - anyway, better to call it something specific such as 
Action.

Comment (hpohekar) [reply src/ansys/fluent/core/solver/flobject.py]: Done. Thank you.

File: src/ansys/fluent/core/solver/flobject.py [hunk 2]
```diff
@@ -679,19 +679,19 @@ def _get_new_keywords(obj, kwds):
     return newkwds
 
 
-class IBase(Base):
+class ActionBase(Base):
     """Intermediate Base class for Command and Query class."""
 
     def __init__(self, name: str = None, parent=None):
-        """__init__ of IBase class."""
+        """__init__ of ActionBase class."""
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/solver/flobject.py:686]: Action is sufficient.

Comment (hpohekar) [reply src/ansys/fluent/core/solver/flobject.py]: Done. Thank you.

**Review by seanpearsonuk — APPROVED** [review]

File: src/ansys/fluent/core/solver/flobject.py [hunk 3]
```diff
@@ -679,56 +679,62 @@ def _get_new_keywords(obj, kwds):
...
+        """__init__ of Action class."""
         super().__init__(name, parent)
         if hasattr(self, "argument_names"):
             for argument in self.argument_names:
                 cls = getattr(self.__class__, argument)
                 self._setattr(argument, cls(None, self))
 
-    def arguments(self) -> Any:
-        """Get the arguments for the command."""
+    def arguments(self, class_name) -> Any:
```
Comment (h-krishnan) [inline src/ansys/fluent/core/solver/flobject.py:693]: No need to pass the `class_name` argument here.  `self.__class__.__name__` should work, and so there is no need for this method in the derived class.

File: src/ansys/fluent/core/solver/flobject.py [hunk 4]
```diff
@@ -679,56 +679,62 @@ def _get_new_keywords(obj, kwds):
...
         attrs = self.get_attrs(["arguments"])
         if attrs:
             attrs = attrs.get("attrs", attrs)
         if attrs and attrs.get("active?", True) is False:
-            raise RuntimeError("Command is not active")
+            raise RuntimeError(f"{class_name} is not active")
         return attrs["arguments"] if attrs else None
 
-    def __call__(self, **kwds):
+    def __call__(self, func_handle, **kwds):
```
[the remark this answers is not in the document]
Comment (hpohekar) [reply src/ansys/fluent/core/solver/flobject.py]: @h-krishnan Done. Thank you.

File: src/ansys/fluent/core/solver/flobject.py (hunk 3 above)
Comment (hpohekar) [reply src/ansys/fluent/core/solver/flobject.py]: Done. Thank you.

**Review by mkundu1 — APPROVED** [review]

File: src/ansys/fluent/core/solver/flobject.py [hunk 5]
```diff
@@ -679,51 +679,46 @@ def _get_new_keywords(obj, kwds):
...
         if attrs and attrs.get("active?", True) is False:
-            raise RuntimeError("Command is not active")
+            raise RuntimeError(f"{self.__class__.__name__} is not active")
         return attrs["arguments"] if attrs else None
 
+
+class Command(Action):
+    """Command object."""
+
+    def __init__(self, name: str = None, parent=None):
```
Comment (h-krishnan) [inline src/ansys/fluent/core/solver/flobject.py:706]: I think these functions are redundant. What is "str" for?

File: src/ansys/fluent/core/solver/flobject.py [hunk 6]
```diff
@@ -679,51 +679,46 @@ def _get_new_keywords(obj, kwds):
...
+        """Call a query with the specified keyword arguments."""
         newkwds = _get_new_keywords(self, kwds)
         return self.flproxy.execute_cmd(self._parent.path, self.obj_name, **newkwds)
 
 
-class Query(Base):
+class Query(Action):
     """Query object."""
 
     def __init__(self, name: str = None, parent=None):
```
Comment (h-krishnan) [inline src/ansys/fluent/core/solver/flobject.py:719]: Is this function redundant? What is `str` for?

**Review by prmukherj — APPROVED** [review]

**Review by h-krishnan — APPROVED** [review]

File: src/ansys/fluent/core/solver/flobject.py (hunk 5 above)
Comment (hpohekar) [reply src/ansys/fluent/core/solver/flobject.py]: Yes, right. Removed. Thank you.

File: src/ansys/fluent/core/solver/flobject.py (hunk 6 above)
Comment (hpohekar) [reply src/ansys/fluent/core/solver/flobject.py]: Yes, right. Removed. Thank you.

**Pull request merged**